classes/core/PKPPageRouter.php

```diff
@@ -399,81 +400,45 @@ public function redirectHome(PKPRequest $request): void
+                    ->where(fn (Builder $q) => $q->whereNull('date_start')->orWhere('date_start', '<=', now()))
+            )
+            ->get();
+        if ($userGroups->isEmpty() || ($userGroups->count() == 1 && $userGroups->first()->role_id == Role::ROLE_ID_READER)) {
+            return $request->url(null, 'index');
+        }
 
-        if ($context = $this->getContext($request)) {
-            // fetch user groups for the user in the current context
-            $userGroups = UserGroup::query()
-                ->where('context_id', $context->getId())
-                ->whereHas('userUserGroups', function ($query) use ($userId) {
-                    $query->where('user_id', $userId)
-                        ->where(function ($q) {
-                            $q->whereNull('date_end')
-                                ->orWhere('date_end', '>', now());
-                        })
-                        ->where(function ($q) {
-                            $q->whereNull('date_start')
-                                ->orWhere('date_start', '<=', now());
-                        });
-                })
-                ->get();
-
-            if ($userGroups->isEmpty()
-                || ($userGroups->count() == 1 && $userGroups->first()->role_id == Role::ROLE_ID_READER)
-            ) {
-                return $request->url(null, 'index');
-            }
-
-            if (Config::getVar('features', 'enable_new_submission_listing')) {
-
-                $roleIdsArray = $userGroups->pluck('role_id')->all();
+        if (Config::getVar('features', 'enable_new_submission_listing')) {
+            $roleIdsArray = $userGroups->pluck('role_id')->all();
 
-                if (array_intersect([Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN, Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_ASSISTANT], $roleIdsArray)) {
-                    return $request->url(null, 'dashboard', 'editorial');
+            if (array_intersect([Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN, Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_ASSISTANT], $roleIdsArray)) {
+                return $request->url(null, 'dashboard', 'editorial');
+            }
 
-                }
-                if (in_array(Role::ROLE_ID_REVIEWER, $roleIdsArray)) {
-                    return $request->url(null, 'dashboard', 'reviewAssignments');
-                }
-                if (in_array(Role::ROLE_ID_AUTHOR, $roleIdsArray)) {
-                    return $request->url(null, 'dashboard', 'mySubmissions');
-                }
+            if (in_array(Role::ROLE_ID_REVIEWER, $roleIdsArray)) {
+                return $request->url(null, 'dashboard', 'reviewAssignments');
             }
 
-            return $request->url(null, 'submissions');
-        } else {
-            // The user is at the site context
-            $userGroups = UserGroup::query()
-                ->where('context_id', \PKP\core\PKPApplication::SITE_CONTEXT_ID)
```

Comment:
Here the query is attempting to find a user group connected to the site context (probably only an item will be retrieved, if the user is the site administrator).